src/main/java/org/prebid/server/handler/CookieSyncHandler.java

```diff
@@ -81,66 +95,127 @@ public CookieSyncHandler(long defaultTimeout,
+        try {
+            bodyAsObjectNode = mapper.decodeValue(body, ObjectNode.class);
+        } catch (DecodeException e) {
+            logger.info(REQUEST_BODY_CANNOT_BE_PARSED, e);
+            throw new InvalidCookieSyncRequestException(REQUEST_BODY_CANNOT_BE_PARSED);
+        }
+
+        return normalizeGppSid(bodyAsObjectNode);
+    }
+
+    private static ObjectNode normalizeGppSid(ObjectNode requestNode) {
+        final JsonNode gppSidNode = requestNode.get(GPP_SID);
+        if (gppSidNode == null) {
+            return requestNode;
         }
 
+        if (!gppSidNode.isTextual()) {
+            logger.info(INVALID_GPP_SID);
+            throw new InvalidCookieSyncRequestException(INVALID_GPP_SID);
+        }
+
+        final List<Integer> gppSid = parseGppSid(gppSidNode.textValue());
+
+        final ArrayNode gppSidArrayNode = requestNode.putArray(GPP_SID);
+        gppSid.forEach(gppSidArrayNode::add);
+
+        return requestNode;
+    }
+
+    private static List<Integer> parseGppSid(String gppSid) {
         try {
-            return mapper.decodeValue(body, CookieSyncRequest.class);
-        } catch (DecodeException e) {
-            final String message = "Request body cannot be parsed";
-            logger.info(message, e);
-            throw new InvalidCookieSyncRequestException(message);
+            return Arrays.stream(gppSid.split(","))
+                    .map(StringUtils::strip)
+                    .filter(StringUtils::isNotBlank)
+                    .map(Integer::parseInt)
+                    .toList();
+        } catch (NumberFormatException e) {
+            logger.info(INVALID_GPP_SID, e);
+            throw new InvalidCookieSyncRequestException(INVALID_GPP_SID);
         }
     }
 
+    private CookieSyncRequest parseRequest(ObjectNode bodyAsObjectNode) {
+        try {
+            return mapper.mapper().treeToValue(bodyAsObjectNode, CookieSyncRequest.class);
+        } catch (JsonProcessingException e) {
+            logger.info(REQUEST_BODY_CANNOT_BE_PARSED, e);
+            throw new InvalidCookieSyncRequestException(REQUEST_BODY_CANNOT_BE_PARSED);
+        }
+    }
+
+    private Future<CookieSyncContext> fillWithAccount(CookieSyncContext cookieSyncContext) {
+        return accountById(cookieSyncContext.getCookieSyncRequest().getAccount(), cookieSyncContext.getTimeout())
+                .map(cookieSyncContext::with);
+    }
+
     private Future<Account> accountById(String accountId, Timeout timeout) {
         return StringUtils.isBlank(accountId)
                 ? Future.succeededFuture(Account.empty(accountId))
                 : applicationSettings.getAccountById(accountId, timeout)
                 .otherwise(Account.empty(accountId));
     }
 
+    private CookieSyncContext processGpp(CookieSyncContext cookieSyncContext) {
+        return cookieSyncContext.with(
+                gppProcessor.apply(cookieSyncContext.getCookieSyncRequest(), cookieSyncContext));
+    }
```

Comment:
I think this should be a part of PrivacyEnforcementService, since its more related to privacy then to parsing request in general. What I am trying to say is - lets keep handler as thin as possible.